packages/jaeger-ui/src/components/Monitor/ServicesView/operationDetailsTable/index.test.js

```diff
@@ -12,291 +12,202 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import '@testing-library/jest-dom';
 import React from 'react';
-import { mount, shallow } from 'enzyme';
-import OperationTableDetails from '.';
+import { render, fireEvent } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { OperationTableDetails } from '.';
 import { originInitialState, serviceOpsMetrics } from '../../../../reducers/metrics.mock';
 import * as track from './index.track';
 
-const props = {
+// Mock LoadingIndicator component since it's a separate import
+jest.mock('../../../common/LoadingIndicator', () => ({
+  __esModule: true,
+  default: () => <div data-testid="loading-indicator">Loading...</div>,
+}));
+
+const defaultProps = {
   data: originInitialState.serviceOpsMetrics,
   error: originInitialState.opsError,
   loading: true,
   endTime: 1632133918915,
   lookback: 3600 * 1000,
   serviceName: 'serviceName',
-  hoveredRowKey: [],
 };
 
-describe('<OperationTableDetails>', () => {
-  let wrapper;
-
+describe('OperationTableDetails', () => {
   beforeEach(() => {
-    wrapper = shallow(<OperationTableDetails {...props} />);
+    jest.spyOn(track, 'trackSortOperations');
+    jest.spyOn(track, 'trackViewTraces');
   });
 
-  it('does not explode', () => {
-    expect(wrapper.length).toBe(1);
+  afterEach(() => {
+    jest.clearAllMocks();
   });
 
-  it('Loading indicator is displayed', () => {
-    expect(wrapper).toMatchSnapshot();
+  it('shows loading indicator when loading', () => {
+    const { getByTestId } = render(<OperationTableDetails {...defaultProps} />);
+    expect(getByTestId('loading-indicator')).toBeInTheDocument();
   });
 
-  it('"Couldn’t fetch data" displayed', () => {
+  it('shows error message when API fails', () => {
     const error = {
       opsCalls: new Error('API Error'),
       opsErrors: new Error('API Error'),
       opsLatencies: new Error('API Error'),
     };
-    wrapper.setProps({ ...props, loading: false, error });
-    expect(wrapper).toMatchSnapshot();
-  });
-
-  it('Table rendered successfully', () => {
-    wrapper.setProps({ ...props, loading: false });
-    expect(wrapper).toMatchSnapshot();
-  });
-});
-
-describe('<OperationTableDetails>', () => {
```

Comment:
I don't understand why there are so many massive changes here. I expect the unit tests to remain structurally the same (especially the test names, to anchor the diff), and only their content to change from `toMatchSnapshot` to explicit checks.